src/pages/_app.tsx

```diff
@@ -27,6 +28,25 @@ if (typeof window !== 'undefined' && process.env.AXE_ENABLED) {
 
 export default function App({ Component, pageProps, layoutProps }) {
   useAnchorLinkAnalytics()
+  const currentPath = useCurrentPath({ excludeHash: true, excludeSearch: true })
+
+  /**
+   * TODO: this will be temporary and last for as long as we have to manually
+   * hydrate product data from page components. Ideally, we won't have to do
+   * this we use more dynamic routes.
+   */
+  if (process.env.HASHI_ENV !== 'production') {
+    const isDevDotPage =
+      !currentPath.startsWith(`/_proxied-dot-io`) &&
+      !currentPath.startsWith(`/swingset`)
```

Comment:
@ashleemboyer I don't think `currentPath` will include `/_proxied-dot-io` when we apply the rewrites, so this check fires when viewing the preview in the context of an io site.